DependencyInjection/Compiler/AddVotersPass.php

```diff
@@ -22,14 +22,28 @@ public function process(ContainerBuilder $container)
         $definition = $container->getDefinition('knp_menu.matcher');
         $listener = $container->getDefinition('knp_menu.listener.voters');
 
-        foreach ($container->findTaggedServiceIds('knp_menu.voter') as $id => $tags) {
-            $definition->addMethodCall('addVoter', array(new Reference($id)));
+        $voters = array();
 
+        foreach ($container->findTaggedServiceIds('knp_menu.voter') as $id => $tags) {
             foreach ($tags as $tag) {
+                $priority = isset($tag['priority']) ? (int) $tag['priority'] : 0;
+                $voters[$priority][] = $id;
+
                 if (isset($tag['request']) && $tag['request']) {
                     $listener->addMethodCall('addVoter', array(new Reference($id)));
                 }
             }
         }
+
+        if (empty($voters)) {
+            return;
+        }
+
+        krsort($voters);
+        $sortedVoters = call_user_func_array('array_merge', $voters);
+
+        foreach ($sortedVoters as $id) {
+            $definition->addMethodCall('addVoter', array(new Reference($id)));
```

Comment:
in the cmf chain router for example we put the priority handling into the chain router. this allows to add more voters at runtime, at the cost of having to sort them at runtime. see https://github.com/symfony-cmf/Routing/blob/master/ChainRouter.php#L80-L93 and https://github.com/symfony-cmf/Routing/blob/master/ChainRouter.php#L124-L134

both have pro and con. is it a realistic scenario that i want to add my voter in a separate compiler pass or at runtime?